File: typescript/base-hmi-types/src/screens/widgets/HmiSymbolicIOField.ts
import { HmiProperty } from "../base.js";
import { HmiTextWidgetBase } from "./HmiTextWidgetBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiState } from "./HmiState.js";
import { HmiStateTriggerMode } from "./HmiStateTriggerMode.js";

export class HmiSymbolicIOField extends HmiTextWidgetBase {
  readonly states: HmiState[] = [];
  configuredStateCount?: HmiProperty<number>;
  stateTriggerMode?: HmiProperty<HmiStateTriggerMode>;
  value?: HmiProperty<number>;
  messageFile?: HmiProperty<string>;
  useVariableMessageFile?: HmiProperty<boolean>;
  useEchoMessage?: HmiProperty<boolean>;
  echoMessage?: HmiProperty<string>;

  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiSymbolicIOField;
  }

  mode?: HmiProperty<number>;
  showDropDownButton?: HmiProperty<boolean>;
  showDropDownList?: HmiProperty<boolean>;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiSymbolicIOField.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public class HmiSymbolicIOField : HmiTextWidgetBase
{
    public HmiSymbolicIOField()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiSymbolicIOField;
    }

    public IList<HmiState> States { get; } = new List<HmiState>();

    /// <summary>
    /// Gets or sets the number of configured states, excluding any error state.
    /// </summary>
    public HmiProperty<int>? ConfiguredStateCount { get; set; }

    public HmiProperty<HmiStateTriggerMode>? StateTriggerMode { get; set; }

    public HmiProperty<double>? Value { get; set; }

    public HmiProperty<string>? MessageFile { get; set; }

    public HmiProperty<bool>? UseVariableMessageFile { get; set; }

    public HmiProperty<bool>? UseEchoMessage { get; set; }

    public HmiProperty<string>? EchoMessage { get; set; }

    public HmiProperty<int>? Mode { get; set; }

    public HmiProperty<bool>? ShowDropDownButton { get; set; }

    public HmiProperty<bool>? ShowDropDownList { get; set; }
}
